src/index.js

```diff
@@ -42,15 +42,18 @@ const runPredicate = ([predicate, errorMsg]:[Function, string],
  * @param {Function} successFn callback function in case of valid input
  * @param {Function} failFn callback function in case of invalid input
  * @param {Object} spec the rule object
- * @param {Object} input the validation input data
+ * @param {Object|Function} input the validation input data
  * @returns {{}}
  */
-export const validate = curry((successFn: Function, failFn: Function, spec: Object, input: Object): Object =>
-  reduce((result, key) => {
-    const value = input[key]
+export const validate = curry((successFn: Function, failFn: Function, spec: Object, input: Object): Object => {
+  const inputFn = typeof input === 'function' ? input : (key?: string) => key ? input : input
```

Comment:
`input : (key?: string) => key ? input : input`

Reply:
It's clearer in the test. `key` is optional as you can also provide a thunk or bound fn that ignores `key`.

Reply:
Ok, let's keep the `key ? input : input` part, I don't mind and adapt when needed. Just want to make sure, it's not confusing to people reading the code.

Reply:
Will merge tomorrow and release a new version. Thanks for all the work @AutoSponge!